feature_details.md

```diff
@@ -0,0 +1,174 @@
+
+* It’s unclear how much painted content for an element should be captured. A shared element could be partially or completely clipped using a CSS clip/clip-path on the element or by an ancestor up to the viewport. However more content may be exposed during the transition TODO: Add example. Since the end state is not known until the new Document loads, this decision needs to be made before the animations for the transition are finalized. The size of the painted content is referred to as “paint bounds” in the rest of the document.
+
+* An element in the subtree of a shared element could have a backdrop-filter which was applied to an ancestor of the shared element TODO: Add example. One option is to make the shared element the backdrop root for any element in its subtree when snapshotting it’s content, effectively flattening the backdrop-filter into the shared element’s snapshot.
+
+##### Animatable Properties
+The properties which can be animated during the transition. Each shared element creates a stacking context and the properties which are animated should closely align with effects which are applied to the stacking context output, as opposed to the painted element. The specific property types animatable with this feature are : opacity, filter, clip, transform, clip-path, mask-image, backdrop-filter. The properties which define how an element blends into its parent stacking context (for example mix-blend-mode) are also retained.
+
+While the properties above map directly to an existing CSS concept, we need to define a new concept for paired transitions. Animating the painted content between the elements involves the following animations. In future iterations, we’d like to expose these to the developer for low level customization, either grouped into a single property (`content`) or exposed separately.
+
+* The container size for the element animates from the paint bounds of the previous element to the paint bounds of the new element. During this animation, neither element’s content fits the size of the container. Fitting the painted content within the container is similar to a resize operation to fit a [replaced element’s](https://developer.mozilla.org/en-US/docs/Web/CSS/Replaced_element) content. The default animation uses the behaviour defined by fill mode in [object-fit](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) but customizing this should be supported going forward. TODO : Add example.
+
+* The pixels from the painted content of the 2 elements are blended using a [cross-fade](https://developer.mozilla.org/en-US/docs/Web/CSS/cross-fade()) animation from `cross-fade(previous 100%, new 0%)` to `cross-fade(previous 0%, new 100%)`.
```

Comment:
I'm unsure about the level of detail in these descriptions. This sounds like we must use the cross-fade algorithm as specified. If the algorithm is changed, then our implementation should change as well. Is this the intent of this text?

The problem is that the blending would typically happen in a different code location / rendering pipeline, so I'm not sure how faithfully we can implement the exact cross-fade algorithm (or rather how much it matters)

Reply:
Thanks for this question!

The main intent here is that the blending function used to combine the 2 images must ensure that blending 2 pixels with the same value is a no-op, or B(a, a) = a. CSS cross-fade's  implementation already does so I was trying to build on the existing concept of how to blend 2 images. I looked at its spec now : https://drafts.csswg.org/css-images-4/Overview.bs, and even that doesn't describe the exact blending operation the implementation must use.

It sounds reasonable to clarify the intent here for now and we can decide the level of detail that must be addressed here based on spec discussions?